docs/docs/troubleshooting-common-errors.md

```diff
@@ -0,0 +1,221 @@
+
+## Errors using gatsby-image and sharp
+
+Gatsby's image processing is broken up into different packages which need to work together to source images and transform them into different optimized versions. You might run into these errors getting them to play together nicely.
+
+### Field "image" must not have a selection since type "String" has no subfields
+
+This errror message `Field "image" must not have a selection since type "String" has no subfields.` comes up when a GraphQL query is trying to query a field for subfields, but none exist. This generally happens when plugins that are used together are added in the `gatsby-config` in the wrong order, or haven't been added at all.
+
+The query is trying to access fields that don't exist because they weren't set up at build time. In the following code, a query is looking to find the subfield `childImageSharp` of the `image` field, like the error states. The problematic GraphQL schema looks like this:
+
+```graphql
+allMdx {
+  nodes {
+    id
+    title
+    image
+  }
+}
+```
+
+And the expected GraphQL schema looks this:
+
+```graphql
+allMdx {
+  nodes {
+    id
+    title
+    image {
+      childImageSharp {
+        fluid {
+          srcSet
+        }
+      }
+    }
+  }
+}
+```
+
+In the first code example, the `image` field was not modified (using Gatsby terminology, transformed) by a plugin to add subfields, so it would only return a string. `gatsby-plugin-sharp` and `gatsby-transformer-sharp` can be included before other plugins that would manipulate or create image nodes (like `gatsby-source-filesystem` or `gatsby-source-contentful`) to ensure that they are present before Gatsby tries to modify them and add the needed fields like `childImageSharp`.
+
+Another possibility is that empty strings are used for image paths somewhere in your site, and when Gatsby constructs a GraphQL schema it [infers](/docs/glossary#inference) the wrong type because the empty string doesn't look like a path.
+
+### Problems installing `sharp` with `gatsby-plugin-sharp` - gyp ERR! build error
+
+If you see an error message in the console when installing dependencies that look related to sharp like `gyp ERR! build error` and `npm ERR! Failed at the sharp@x.x.x install script`, they can often be resolved by deleting the `node_nodules` folder in the root of your project and installing dependencies again:
+
+```shell
+# be careful as this command will delete all files recursively in
+# the folder you provide, in this case, node_modules
+rm -rf node_modules
+
+# this command will install libraries from your package.json file
+# and place them in the node_modules folder
+npm install
+```
+
+The version of Node.js that's used to install sharp needs to match the version of Node.js that is run, so clearing `node_modules` and reinstalling often resolves the problem.
+
+### Incompatible library version: sharp.node requires version X or later, but Y provides version Z
```

Comment:
This issue could provide more info here: https://github.com/gatsbyjs/gatsby/issues/10841